pytest.ini

```diff
@@ -1,2 +1,3 @@
 [pytest]
 timeout = 1
+addopts = -n 4 --dist loadgroup
```

Comment:
Is there a benefit from not using auto to avoid being slower in a machine with 2 cores and could underutilize a machine with many core? 
If we switch to auto, we could also probably remove the argument in `.github/workflows/unit-tests.yml`?

Reply:
In the case of your specific test suite, no I don't think we'd underutilize a machine with many cores. There are enough tests that require multiple cores that there wasn't really an observable speedup. See my notes under `Numbers` in the PR.

Additionally, my understanding is that, when given auto, `pytest-xdist` will create as many workers as you have logical cores. In the case that your test suite did parallelize super well, we'd run into an issue with the ordering of the test groups. Tests which use multiple cores would need to be run last to avoid resource starvation while every other core is executing a test. If we didn't handle that, some might fail because they would exceed their time limit.